extension/js/common/api/shared/api.ts

```diff
@@ -412,7 +412,24 @@ export class Api {
       const result = await Api.ajaxWithJquery(req, resFmt, formattedData);
       return result as FetchResult<T, RT>;
     } else {
-      return await Api.ajax(req, resFmt);
+      try {
+        return await Api.ajax(req, resFmt);
+      } catch (firstAttemptErr) {
+        const idToken = headers?.authorization?.split('Bearer ')?.[1];
+        if (ApiErr.isAuthErr(firstAttemptErr) && idToken) {
+          // Needed authorization from the service worker side to avoid circular dependency injection errors
+          // that occur when importing GoogleAuth directly.
+          const authorization = await BrowserMsg.send.bg.await.getGoogleApiAuthorization({ idToken });
```

Comment:
maybe it's possible to fix circular dependency without using service worker?
as it's better to keep service worker lightweight, and use it only for tasks which are difficult or not possible to be implemented in the content script.

Reply:
I attempted to resolve the circular dependency issue by creating a base service, such as `HttpService`, which contains only the essential common functions. I then had the `GoogleOAuth` and `Api` classes extend this base service. To circumvent the circular dependency, I used the `GoogleOAuth.googleApiAuthHeader` function in a `catch` block, thinking this separation would resolve the issue since the `GoogleOAuth` and `Api` classes wouldn't be directly related. However, the circular dependency persisted because the `Api` class still utilizes the `GoogleOAuth` class, and `GoogleOAuth` imports the `AccountServer` class, which, in turn, extends the `Api` class.

In my opinion, solving this circular dependency issue would require significant changes, which I find a bit overwhelming and am uncertain if the effort would be justified.

Reply:
Let me know your thoughts

Reply:
How can I reproduce such circular dependency error?
I tried to build extension from your commit before circular dependency fix, builds passed without any errors and extension seems to work well.

Reply:
Was able to reproduce by running failed tests, will check it